### PR TITLE
Karla rubiano NSCache and os.log

### PR DESCRIPTION
**Pull request by kdrubianoc**
#### What does this PR do?
Feat: Remove KingFisher interface and use NSCache. Use os.log to log errors.
Enhancement: Added an alert if the movie doesn't have reviews. 

#### Where should the reviewer start?
UIImageView+Extensions, ViewModels files, and ViewControllers files.

#### Code review checklist [Code Review Rubric](https://github.com/wizeline/iOS-apprenticeship-program-2020/blob/main/code-review-rubric.md)

1. DRY Principle (Don’t repeat yourself)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

2. Yagni Principle (You Aren't Gonna Need It)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

3. KISS Principle (Keep It Simple, St...d)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

4. Remove unused code

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

5. Hard-coded values

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

6. Language style guide & coding conventions
 
 - [Wizeline Style Guide](https://github.com/wizeline/mobile-documentation/blob/master/guidelines/swift-style-guide.md) 
 - [RayWenderlich Style Guide](https://github.com/raywenderlich/swift-style-guide)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

7. Architecture / Design Pattern

    - [ ] MVC  
    - [ ] MVP 
    - [ ] MVVM 
    - [ ] VIPER 

8. Architecture Implementation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

9. SOLID

- Single Responsibility Principle 

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Open Closed Principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Liskov substitutability principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Interface segregation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Dependency Inversion principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

10. Testing

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

11. Force unwrapping !

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

12. Mutable vs immutable property (Right use of `let`  and `var`)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

13. Retain cycles (Capture list)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

14. Code Documentation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

### CODE REVIEW

File: TheMovieDb/Helpers/Constants.swift [hunk 1]
```diff
@@ -24,6 +24,7 @@ struct Constants {
     static let titleInitialTableView = "Show Movies"
     static let searchBarPlaceholder = "Search for movies or person"
     static let alertButton = "Dismiss"
+    static let alertEmptyReviews = "Reviews Not Found"
```
Comment (Carlos-Barron) [inline TheMovieDb/Helpers/Constants.swift:27]: Remember that if you haver Localizable files, you can move these titles to the file and only get the reference.

File: TheMovieDb/Helpers/UIImageView+Extensions.swift [hunk 2]
```diff
@@ -7,15 +7,35 @@
...
+                let httpURLResponse = response as? HTTPURLResponse, httpURLResponse.statusCode == 200,
+                let mimeType = response?.mimeType, mimeType.hasPrefix("image"),
+                let data = data, error == nil,
+                let imageToCache = UIImage(data: data)
+            else { return }
+            DispatchQueue.main.async() {
+                self.image = imageToCache
+                imageCache.setObject(imageToCache, forKey: url as AnyObject)
+            }
+        }.resume()
```
Comment (Carlos-Barron) [inline TheMovieDb/Helpers/UIImageView+Extensions.swift:39]: Nice implementation and the usage of GCD.

**Review by Carlos-Barron — APPROVED** [review]
LGTM! Just a comment about the localizable files

File: TheMovieDb/Helpers/Constants.swift (hunk 1 above)
Comment (kdrubianoc) [reply TheMovieDb/Helpers/Constants.swift:27]: Yes, I'm going to implement a localizable file in the next PR.

**Pull request merged**